src/components/Chat/InputMessageV2.tsx

```diff
@@ -0,0 +1,219 @@
+  containerInput: {
+    flex: 1,
+    backgroundColor: colors.lightgrey,
+    flexDirection: 'row',
+    alignItems: 'flex-end',
+    paddingVertical: dimen.normalizeDimen(6),
+    paddingHorizontal: dimen.normalizeDimen(8),
+    borderRadius: dimen.normalizeDimen(12),
+    minHeight: dimen.normalizeDimen(32),
+    marginHorizontal: dimen.normalizeDimen(6)
+  },
+  input: {
+    flex: 1,
+    backgroundColor: colors.lightgrey,
+    color: colors.black,
+    fontFamily: 'Inter',
+    fontSize: normalizeFontSize(14),
+    fontStyle: 'normal',
+    fontWeight: '400',
+    marginRight: dimen.normalizeDimen(12),
+    minHeight: dimen.normalizeDimen(20),
+    maxHeight: dimen.normalizeDimen(80),
+    textAlignVertical: 'center',
+    paddingTop: dimen.normalizeDimen(2)
+  },
+  btn: {
+    borderRadius: dimen.normalizeDimen(32 / 2),
+    width: dimen.normalizeDimen(32),
+    height: dimen.normalizeDimen(32),
+    display: 'flex'
+  },
+  enableButton: {
+    backgroundColor: colors.bondi_blue
+  },
+  disableButton: {
+    backgroundColor: colors.gray1
+  },
+  previewPhotoContainer: {
+    marginTop: dimen.normalizeDimen(5),
+    marginBottom: dimen.normalizeDimen(5)
+  },
+  imageStyle: {
+    height: dimen.normalizeDimen(64),
+    width: dimen.normalizeDimen(64),
+    marginRight: dimen.normalizeDimen(10)
+  },
+  containerDelete: {
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    height: dimen.normalizeDimen(25),
+    width: dimen.normalizeDimen(25),
+    backgroundColor: colors.holytosca,
+    alignItems: 'center',
+    justifyContent: 'center',
+    zIndex: dimen.normalizeDimen(100),
+    borderRadius: dimen.normalizeDimen(13)
+  },
+  labelToggle: {
+    fontSize: dimen.normalizeDimen(8),
+    fontWeight: '400'
+  }
+});
+
+export interface InputMessageV2Props {
+  onSendButtonClicked: (message: string) => void;
+  type: 'SIGNED' | 'ANONYMOUS';
+  emojiCode?: string;
+  emojiColor?: string;
+  username?: string;
+  profileImage?: string;
+  onToggleConfirm?: () => void;
+  isShowToggle?: boolean;
+  messageDisable?: string | null;
+}
+
+const InputMessageV2 = ({
+  onSendButtonClicked,
+  emojiCode,
+  emojiColor,
+  profileImage,
+  username,
+  onToggleConfirm,
+  type,
+  isShowToggle = true,
+  messageDisable
+}: InputMessageV2Props) => {
+  const refEmoji = React.useRef(null);
+  const [inputFocus, setInputFocus] = React.useState(false);
+  const [text, setText] = React.useState('');
+
+  const onChangeInput = (v) => {
+    setText(v);
+  };
+  const onSelectEmoji = () => {
+    refEmoji.current.close();
+  };
+
+  const handleSendMessage = () => {
+    onSendButtonClicked(text);
+    setText('');
+  };
+
+  const isDisableButton = () => {
+    return text?.length === 0;
+  };
+
+  const sendButtonStyle = React.useCallback(() => {
+    const isDisabled = isDisableButton();
+    if (isDisabled) return colors.gray1;
+    if (type === 'SIGNED') return colors.darkBlue;
+    return colors.bondi_blue;
+  }, [isDisableButton()]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendButtonStyle` function uses `isDisableButton()` inside a `useCallback` dependency array. This could lead to unexpected behavior because `isDisableButton()` is a function that returns a boolean value based on the current state of `text`. Instead, you should include `text` directly in the dependency array. This will ensure that the callback is updated whenever the `text` state changes.

```diff
-  const sendButtonStyle = React.useCallback(() => {
-    const isDisabled = isDisableButton();
-    if (isDisabled) return colors.gray1;
-    if (type === 'SIGNED') return colors.darkBlue;
-    return colors.bondi_blue;
-  }, [isDisableButton()]);
+  const sendButtonStyle = React.useCallback(() => {
+    const isDisabled = text?.length === 0;
+    if (isDisabled) return colors.gray1;
+    if (type === 'SIGNED') return colors.darkBlue;
+    return colors.bondi_blue;
+  }, [text, type]);
```
This change ensures that the `sendButtonStyle` function is re-computed when either `text` or `type` changes, which is the intended behavior.


<!-- This is an auto-generated comment by OSS CodeRabbit -->